### PR TITLE
docs(no-redeclare): add docs

### PR DESCRIPTION
**Pull request by magurotuna**
This PR adds documentation for `no-redeclare`.

### CODE REVIEW

File: src/rules/no_redeclare.rs [hunk 1]
```diff
@@ -40,6 +40,49 @@ impl LintRule for NoRedeclare {
       ProgramRef::Script(ref s) => visitor.visit_script(s, &DUMMY_NODE),
     }
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Disallows redeclaration of variables.
```
Comment (dsherret) [inline src/rules/no_redeclare.rs:45]: This rule seems to affect also parameter, function, and class property declarations. Perhaps this sentence should be changed?

Comment (magurotuna) [reply src/rules/no_redeclare.rs:45]: Good point, I missed those! I'll fix it today.

Comment (magurotuna) [reply src/rules/no_redeclare.rs]: @dsherret Done, please take a look again.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks @magurotuna!

**Pull request merged**